src/Events/Services/PushEventService.cs

```diff
@@ -183,34 +149,36 @@ private static CloudEventEnvelope MapToEnvelope(CloudEvent cloudEvent, Subscript
             return cloudEventEnvelope;
         }
 
-        private static string GetOrgAppSubscriptionCacheKey(string source, string subject, string type)
+        private static string GetSubscriptionCacheKey(string source, string subject, string type)
         {
             if (source == null)
             {
                 return null;
             }
 
-            return "orgsubscription:so:" + source + "su:" + subject + "ty:" + type;
+            return "subscription:so:" + source + "su:" + subject + "ty:" + type;
         }
 
-        private static string GetPartyAppSubscriptionCacheKey(string source, string subject, string type)
+        private static string GetAltinnAppAuthorizationCacheKey(string sourceFilter, string consumer)
         {
-            if (source == null)
+            if (sourceFilter == null)
             {
                 return null;
             }
 
-            return "partysubscription:so:" + source + "su:" + subject + "ty:" + type;
+            return "authorizationdecision:so:" + sourceFilter + "co:" + consumer;
         }
 
-        private static string GetAltinnAppAuthorizationCacheKey(string sourceFilter, string consumer)
+        private string GetSourceFilter(Uri source)
         {
-            if (sourceFilter == null)
+            if (source.DnsSafeHost.Contains(_platformSettings.AppsDomain))
             {
-                return null;
+                return source.OriginalString.Substring(0, source.OriginalString.IndexOf(source.Segments[3]) - 1);
```

Comment:
Is there a check that source has at least 3 segments somewhere before this is executed, or are we relying on the creation of app events to only create events with org/app/instance sources?

Reply:
No checks on the segment. As the method is only called from AuthorizeConsumerForAltinnAppEvent, we rely on it being n app event for now